src/handlers/int-test-support/helpers/testSetup.ts

```diff
@@ -1,51 +1,76 @@
 import { poll } from "./commonHelpers";
-import { resourcePrefix } from "./envHelper";
-import { deleteS3ObjectsByPrefix, listS3Objects } from "./s3Helper";
-
-const prefix = resourcePrefix();
-const storageBucket = `${prefix}-storage`;
+import {
+  deleteS3Objects,
+  deleteS3ObjectsByPrefix,
+  listS3Objects,
+} from "./s3Helper";
+import {
+  S3_TRANSACTION_FOLDER,
+  STORAGE_BUCKET,
+  S3_INVOICE_FOLDER,
+  S3_INVOICE_ARCHIVED_FOLDER,
+  RAW_INVOICE_BUCKET,
+  RAW_INVOICE_TEXTRACT_BUCKET,
+} from "../test-constants";
 
 export default async function globalSetup(): Promise<void> {
-  // Delete objects with prefix "btm_event_data/2005"
-  await deleteS3ObjectsByPrefix({
-    bucket: storageBucket,
-    prefixes: ["btm_event_data/2005"],
-  });
-  // poll to ensure that the objects with prefix "btm_event_data/2005" have been deleted
+  console.log("Cleaning up the environment before test execution starts");
+  await deleteS3ObjectsAndPoll(STORAGE_BUCKET, `${S3_TRANSACTION_FOLDER}/2005`);
+  await deleteS3ObjectsAndPoll(STORAGE_BUCKET, S3_INVOICE_FOLDER);
+  await deleteS3ObjectsAndPoll(STORAGE_BUCKET, S3_INVOICE_ARCHIVED_FOLDER);
+  await deleteAllObjects(RAW_INVOICE_BUCKET);
+  await deleteAllObjects(RAW_INVOICE_TEXTRACT_BUCKET);
+  console.log("All previous test data has been cleaned.");
+}
+
+const deleteS3ObjectsAndPoll = async (
+  bucketName: string,
+  prefix: string
+): Promise<void> => {
+  await deleteS3ObjectsByPrefix({ bucket: bucketName, prefixes: [prefix] });
+
+  // poll to ensure that the objects with prefix have been deleted
   await poll(
     async () =>
       await listS3Objects({
-        bucketName: storageBucket,
-        prefix: "btm_event_data/2005",
+        bucketName,
+        prefix,
       }),
     (s3Objects) => s3Objects.length === 0,
     {
       timeout: 60000,
       interval: 10000,
-      notCompleteErrorMessage:
-        "btm_event_data/2005 folder could not be deleted because it still contains objects",
+      notCompleteErrorMessage: `${prefix} folder could not be deleted because it still contains objects`,
     }
   );
+};
 
-  // Delete objects with prefix "btm_invoice_data"
-  await deleteS3ObjectsByPrefix({
-    bucket: storageBucket,
-    prefixes: ["btm_invoice_data"],
-  });
+const deleteAllObjects = async (bucketName: string): Promise<void> => {
```

Comment:
This could fail if the list of objects is too big. [This thread](https://stackoverflow.com/questions/30755129/aws-s3-listobjects-with-pagination) has some good examples of how to paginate through the results

Reply:
the listS3ObjectsBasic function in s3Helper , handles the pagination using continuation token.